zipline/utils/serialization_utils.py

```diff
@@ -13,6 +13,66 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from six import BytesIO
+import pickle
+from functools import partial
+
+from zipline.assets import AssetFinder
+from zipline.finance.trading import TradingEnvironment
+
 # Label for the serialization version field in the state returned by
 # __getstate__.
 VERSION_LABEL = '_stateversion_'
+
+
+def _persistent_id(obj):
+    if isinstance(obj, AssetFinder):
+        return AssetFinder.PERSISTENT_TOKEN
+    if isinstance(obj, TradingEnvironment):
+        return TradingEnvironment.PERSISTENT_TOKEN
+    return None
+
+
+def _persistent_load(persid, env):
+    if persid == AssetFinder.PERSISTENT_TOKEN:
+        return env.asset_finder
+    if persid == TradingEnvironment.PERSISTENT_TOKEN:
+        return env
+
+
+def dump_with_persistent_ids(obj, protocol=None):
```

Comment:
Looks right to me!  Naming-wise, if we want to match the schema that pickle uses, these should probably be called ```dumps_with...``` and ```loads_with...``` (with an 's'), since they return/take bytes as opposed to files.